### ISSUE TITLE
openvino python package build failed on windows with rel-1.6.0

### ISSUE THREAD

**Issue by leqiao-1** [microsoft/onnxruntime#7480]
**Describe the bug**
openvino python package build failed on windows with rel-1.6.0

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Microsost Windows 20H2 19042.928
- ONNX Runtime installed from (source or binary): source 
- ONNX Runtime version: 1.6.0
- Python version: Python 3.7.8
- Visual Studio version (if applicable):  Visual Studio 16 2019

**To Reproduce**
- Describe steps/code to reproduce the behavior.
- Attach the ONNX model to the issue (where applicable) to expedite investigation.

I try to build python package for openvino EP on windows by running cmd in onnxruntime repo with branch rel-1.6.0.
```
.\build.bat --config Release --build_shared_lib --use_openmp --use_openvino CPU_FP32   --build_wheel --cmake_generator "Visual Studio 16 2019"
```
It passed serveral tests, but failed later. 
```
6/6 Test #6: onnxruntime_api_tests_without_env ......   Passed    0.08 sec

100% tests passed, 0 tests failed out of 6
```
I have tried with openvino 2020.04 and 2021.01, both failed with error 
```
2021-04-28 10:55:19,530 util.run [INFO] - Running subprocess in 'D:\work\onnxruntime\\build\Windows\Release\Release'
['C:\\Users\\leqiao\\AppData\\Local\\Programs\\Python\\Python37\\python.exe', 'onnxruntime_test_python.py']
2021-04-28 10:55:20.1623568 [E:onnxruntime:Default, provider_bridge_ort.cc:662 onnxruntime::ProviderLibrary::Get] Failed to load library, error code: 126
Traceback (most recent call last):
  File "D:\work\onnxruntime\\tools\ci_build\build.py", line 2088, in <module>
    sys.exit(main())
  File "D:\work\onnxruntime\\tools\ci_build\build.py", line 2026, in main
    run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs)
  File "D:\work\onnxruntime\\tools\ci_build\build.py", line 1466, in run_onnxruntime_tests
    run_subprocess([sys.executable, 'onnxruntime_test_python.py'], cwd=cwd, dll_path=dll_path)
  File "D:\work\onnxruntime\\tools\ci_build\build.py", line 523, in run_subprocess
    return run(*args, cwd=cwd, capture_stdout=capture_stdout, shell=shell, env=my_env)
  File "D:\work\onnxruntime\tools\python\util\run.py", line 42, in run
    env=env, shell=shell)
  File "C:\Users\leqiao\AppData\Local\Programs\Python\Python37\lib\subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['C:\\Users\\leqiao\\AppData\\Local\\Programs\\Python\\Python37\\python.exe', 'onnxruntime_test_python.py']' returned non-zero exit status 3221225477.
```
I have successfully built ort for dnnl ep in same environment. 

**Expected behavior**

**Screenshots**

**Additional context**
```
D:\work\onnxruntime>cmake --version
cmake version 3.20.1
```

**Comment by oliviajain** [comment]
This is likely due to python environment conflicts. It may help to build from scratch and run in a clean conda environment.

**Comment by leqiao-1** [comment]
@oliviajain Thanks for your reply. I tries with a clean conda environment with openvino 2021.2, but it also failed with same error. 

```
(python37_ort) C:\Program Files (x86)\IntelSWTools\openvino_2021.2.185\bin>pip list
Package           Version
----------------- -------------------
appdirs           1.4.4
astroid           2.4.2
certifi           2020.12.5
distlib           0.3.1
isort             5.5.2
lazy-object-proxy 1.4.3
mccabe            0.6.1
pip               21.0.1
pylint            2.6.0
setuptools        52.0.0.post20210125
toml              0.10.1
typed-ast         1.4.1
virtualenv        20.4.2
wheel             0.36.2
wincertstore      0.2
wrapt             1.12.1
```
```
-- ******** Summary ********
--   CMake version         : 3.20.1
--   CMake command         : C:/Program Files/CMake/bin/cmake.exe
--   System                : Windows
--   C++ compiler          : C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe
--   C++ compiler version  : 19.28.29914.0
--   CXX flags             : /DWIN32 /D_WINDOWS /GR /EHsc /W3 -openmp /EHsc /wd26812
--   Build type            : Release
--   Compile definitions   : ENABLE_ORT_FORMAT_LOAD;EIGEN_MPL2_ONLY;EIGEN_HAS_CONSTEXPR;EIGEN_HAS_VARIADIC_TEMPLATES;EIGEN_HAS_CXX11_MATH;EIGEN_HAS_CXX11_ATOMIC;EIGEN_STRONG_INLINE=inline
--   CMAKE_PREFIX_PATH     :
--   CMAKE_INSTALL_PREFIX  : C:/Program Files/onnxruntime
--   CMAKE_MODULE_PATH     : D:/work/onnxruntime/cmake/external
--
--   ONNX version          : 1.8.0
--   ONNX NAMESPACE        : onnx
--   ONNX_BUILD_TESTS      : OFF
--   ONNX_BUILD_BENCHMARKS : OFF
--   ONNX_USE_LITE_PROTO   : OFF
--   ONNXIFI_DUMMY_BACKEND : OFF
--   ONNXIFI_ENABLE_EXT    : OFF
--
--   Protobuf compiler     :
--   Protobuf includes     :
--   Protobuf libraries    :
--   BUILD_ONNX_PYTHON     : OFF
-- Looking for _strtof_l
-- Looking for _strtof_l - found
-- Looking for _strtoui64_l
-- Looking for _strtoui64_l - found
-- The ASM_MASM compiler identification is MSVC
-- Found assembler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/ml64.exe
-- Found NumPy: C:/Users/leqiao/.conda/envs/python37_ort/lib/site-packages/numpy/core/include (found version "1.20.2")
-- NumPy ver. 1.20.2 found (include: C:/Users/leqiao/.conda/envs/python37_ort/lib/site-packages/numpy/core/include)
-- Configuring done
-- Generating done
```
```
Total Test time (real) = 136.64 sec
2021-04-29 16:33:29,575 util.run [DEBUG] - Subprocess completed. Return code: 0
2021-04-29 16:33:29,577 util.run [INFO] - Running subprocess in 'D:\work\onnxruntime\build\Windows\Release\Release'
['C:\\Users\\leqiao\\.conda\\envs\\python37_ort\\python.exe', 'onnxruntime_test_python.py']
2021-04-29 16:33:30.8659072 [E:onnxruntime:Default, provider_bridge_ort.cc:589 onnxruntime::ProviderSharedLibrary::Ensure] Failed to load library, error code: 126
Traceback (most recent call last):
  File "D:\work\onnxruntime\\tools\ci_build\build.py", line 1986, in <module>
    sys.exit(main())
  File "D:\work\onnxruntime\\tools\ci_build\build.py", line 1924, in main
    run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs)
  File "D:\work\onnxruntime\\tools\ci_build\build.py", line 1362, in run_onnxruntime_tests
    run_subprocess([sys.executable, 'onnxruntime_test_python.py'], cwd=cwd, dll_path=dll_path)
  File "D:\work\onnxruntime\\tools\ci_build\build.py", line 528, in run_subprocess
    return run(*args, cwd=cwd, capture_stdout=capture_stdout, shell=shell, env=my_env)
  File "D:\work\onnxruntime\tools\python\util\run.py", line 44, in run
    env=env, shell=shell)
  File "C:\Users\leqiao\.conda\envs\python37_ort\lib\subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['C:\\Users\\leqiao\\.conda\\envs\\python37_ort\\python.exe', 'onnxruntime_test_python.py']' returned non-zero exit status 3221225477.
```

**Comment by MaajidKhan** [comment]
@leqiao-1 can you use the latest ONNX Runtime version: 1.7.0 version and not the 1.6.0 version and follow the same steps.
There was one bug with python API on windows with ONNXRuntime 1.6.0 release. we had it fixed with ONNXRuntime 1.7.0 release. You should probably be able to build and run successfully with ONNXRuntime 1.7.0.

you can use any OpenVINO version (2021.1 also is fine), but if possible, I would recommend you to use the latest OpenVINO 2021.3 Release installation on windows.

**Comment by leqiao-1** [comment]
Thanks @MaajidKhan 
Same error occured with rel-1.7.0
![image](https://user-images.githubusercontent.com/61653207/117289247-f4cbe100-ae9e-11eb-8cf5-a21ae41cc99a.png)

**Comment by MaajidKhan** [comment]
@leqiao-1 I was able to see the same error at the end. This is happening when it tries to run the unit tests automatically once the build is complete. As you see, all the unit tests are passing fine (0 tests failed out of 6 test suite) and so it  looks like there's some issue when it's about to exit while it is trying to run onnxruntime_test_python.py

However you can ignore it for now. The build is anyways successful, you can now navigate down to
cd onnxruntime/build/windows/Release/Release and start experimenting with built in c++ and python applications that comes with onnxruntime (or) you can start writing your own c++ or python application and try executing them.

**Comment by MaajidKhan** [comment]
we also have an option where you can try building onnxruntime and skip the tests.

Just an additional flag **--skip_tests**

You can give it a try. In case, you want to avoid running the tests.
.\build.bat --config Release --build_shared_lib --use_openmp --use_openvino CPU_FP32   --build_wheel --cmake_generator "Visual Studio 16 2019" -**-skip_tests**

Also you can use --parallel to speed up the build process. This flag when used, uses all the cpu cores while building making the overall build time process quick.

.\build.bat --config Release --build_shared_lib --use_openmp --use_openvino CPU_FP32   --build_wheel --cmake_generator "Visual Studio 16 2019" **--parallel --skip_tests**

**Comment by oliviajain** [comment]
Resolved offline. Issue was that .dlls weren't generated in onnxruntime\build\Windows\Release\Release\onnxruntime\capi properly. Copying over the required onnxruntime_providers_(shared/openvino) dlls from the *\Release\Release folder and rebuilding fixed the issue.

**Comment by leqiao-1** [comment]
Thanks for all your help. The problem was resolved. It might be related with 
```
if (onnxruntime_USE_OPENVINO)
  if(NOT WIN32)
    add_custom_command(
      TARGET onnxruntime_pybind11_state POST_BUILD
      COMMAND ${CMAKE_COMMAND} -E copy
          ${OPENVINO_DLL_PATH} $<TARGET_FILE:onnxruntime_providers_openvino>
          $<TARGET_FILE:onnxruntime_providers_shared>
          $<TARGET_FILE_DIR:${build_output_target}>/onnxruntime/capi/
    )
  endif()
endif()
```
in onnxruntime_python.cmake for version early than 1.7.0. It has been changed into 
```
if (onnxruntime_USE_OPENVINO)
    add_custom_command(
      TARGET onnxruntime_pybind11_state POST_BUILD
      COMMAND ${CMAKE_COMMAND} -E copy
          $<TARGET_FILE:onnxruntime_providers_openvino>
          $<TARGET_FILE:onnxruntime_providers_shared>
          $<TARGET_FILE_DIR:${build_output_target}>/onnxruntime/capi/
    )
endif()
```
in master branch and it can work. 

Thanks again.

**Comment by MaajidKhan** [comment]
Glad to hear that. Looks like the fix didn't go into onnxruntime 1.7 release. This  fix will be officially available with the next onnxruntime 1.8 release which is gonna be out after a week or so. For now, you can work with master.

please close the ticket if you don't have any other issues. @leqiao-1
